mtags/src/main/scala/scala/meta/internal/mtags/Enrichments.scala

```diff
@@ -0,0 +1,115 @@
+package scala.meta.internal.mtags
+
+import java.nio.charset.StandardCharsets
+import java.nio.file.Path
+import scala.meta.inputs.Input
+import scala.meta.inputs.Position
+import scala.meta.internal.io.FileIO
+import scala.meta.internal.semanticdb.Language
+import scala.meta.internal.semanticdb.Scala.Descriptor
+import scala.meta.io.AbsolutePath
+import scala.meta.internal.semanticdb.Scala._
+import scala.meta.internal.{semanticdb => s}
+
+object Enrichments {
+  implicit class XtensionDescriptorMetals(desc: Descriptor) {
+    def isCaseClassSynthetic: Boolean = {
+      desc.isMethod && {
+        desc.name.value match {
+          case "apply" | "copy" => true
+          case _ => false
+        }
+      }
+    }
+  }
+
+  implicit class XtensionSymbolMetals(symbol: String) {
+
+    def toplevel: String = {
+      if (symbol.isNone) symbol
+      else if (symbol.isPackage) symbol
+      else {
+        val owner = symbol.owner
+        if (owner.isPackage) symbol
+        else owner.toplevel
+      }
+    }
+    def isToplevel: Boolean = {
+      !symbol.isPackage &&
+      symbol.owner.isPackage
+    }
+  }
+  implicit class XtensionRange(val range: s.Range) {
```

Comment:
why `val`?

Reply:
It was `extends AnyVal` before, I removed the `val`

Reply:
I have no evidence to support that non-`AnyVal` extension methods are any slower so let's keep the ceremony down instead for now.